signals.go

```diff
@@ -61,20 +70,19 @@ type signalHandler struct {
 func (h *signalHandler) forward(process *libcontainer.Process, tty *tty, detach bool) (int, error) {
 	// make sure we know the pid of our main process so that we can return
 	// after it dies.
-	if detach && h.notifySocket == nil {
-		return 0, nil
-	}
-
 	pid1, err := process.Pid()
 	if err != nil {
 		return -1, err
 	}
 
-	if h.notifySocket != nil {
-		if detach {
+	if detach {
+		if h.notifySocket != nil {
 			_ = h.notifySocket.run(pid1)
-			return 0, nil
 		}
+		return 0, nil
```

Comment:
I think the old code that (lines 64-67 deleted) seems simpler. To filter out the cases where we don't need to do anything at the beginning seems more readable, IMHO.

Reply:
Either way is fine with me (or not -- frankly, I find this part code somewhat more complicated than it should be).

Reply:
My thought was that one path is for detached containers, and the rest path is for non-detached containers. So I think it’s more clear than before. If you want, I can remove the last commit.

Reply:
I think maybe it’s more clear to assess if you use the ‘view file’ button to see the whole function body, instead of the difference view.

Reply:
Has moved notifySocket out of signalHandler now.